src/main/java/school/lemon/changerequest/java/io/hw1/FileManager.java

```diff
@@ -0,0 +1,175 @@
+        Arrays.sort(files, new FileComparator(order, directoriesFirst));
+        List<File> fileList = new ArrayList<>();
+        for (File file : files) {
+            fileList.add(file);
+        }
+
+        return fileList;
+    }
+
+    @Override
+    public String fileInfo(File file) {
+        final long lastModified = file.lastModified();
+        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
+        String lastMod = sdf.format(new Date(lastModified));
+        long size;
+        if (file.isDirectory()) {
+            size = 0;
+        } else {
+            size = file.length();
+        }
+        StringBuilder name = new StringBuilder(file.getName());
+        if (file.isDirectory()) name.append("/");
+        String fileInfo = String.format("%-20s | %15d | %s", lastMod, size, name);
+        return fileInfo;
+    }
+
+    @Override
+    public void createFile(String name) throws FileAlreadyExistException, FileManagerException {
+        try {
+            File file = new File(name);
+            if (file.exists()) throw new FileAlreadyExistException(file);
+            file.createNewFile();
+            System.out.println("File has been created.");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void createDirectory(String name) throws DirectoryAlreadyExistException, FileManagerException {
+        File directory = new File(name);
+        if (directory.exists() && directory.isDirectory()) throw new DirectoryAlreadyExistException(directory);
+        try {
+            if (directory.mkdir()) System.out.println("Directory has been created. " + directory.getAbsolutePath());
+        } catch (FileManagerException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void changeDirectory(String name) throws IllegalDirectoryException {
+        File newCurrentDirectory = new File(name).getAbsoluteFile();
+        if (!newCurrentDirectory.exists() || !newCurrentDirectory.isDirectory())
+            throw new IllegalDirectoryException(newCurrentDirectory);
+        currentDirectory = newCurrentDirectory;
+        System.out.println("Current directory has been changed.");
+    }
+
+    @Override
+    public boolean delete(String name) throws FileNotFoundException {
+        boolean delete = delete(name, true);
+        return delete;
+    }
+
+    @Override
+    public boolean delete(String name, boolean force) throws FileNotFoundException {
+        File fileToDelete = new File(name).getAbsoluteFile();
+        if (!fileToDelete.exists()) throw new FileNotFoundException(name);
+        if (fileToDelete.isFile() || fileToDelete.list().length == 0) {
+            fileToDelete.delete();
+            return true;
+        } else {
+            if (force == false) return false;
+            File[] files = fileToDelete.listFiles();
+            for (File f : files) {
+                f.delete();
```

Comment:
1. Please return false in case file wasn't deleted.
2. What about sub-directories?